### ISSUE TITLE
Fullscreen with Gnome-Shell 3.38.1

### ISSUE THREAD

**Issue by Draky50110** [dosbox-staging/dosbox-staging#755]
Hello.

As said "Some users of Gnome 3.38.1 experience problems when toggling fullscreen (both X11 and Wayland). This seems to be a Gnome issue; if you experience it as well, please make a report in the bugtracker." in https://dosbox-staging.github.io/v0-76-0/ I have a problem :)

Ubuntu 20.10 + Gnome-Shell 3.38.1 (X11).

Launching dosbox-staging (from Snap "edge"), if I use ALT+ENTER to enter fullscreen mode, dosbox window is not fullscreen but what is inside is zoomed.
![Capture d’écran de 2020-12-06 20-06-46](https://user-images.githubusercontent.com/1558487/101289788-ddf56580-37fe-11eb-9bae-029b52636c4a.png)

If I use again ALT+ENTER : it does not change. I can use "exit" command on DOS prompt to exit...
If I run a game directly, then ALT+ENTER, problem is here and I can't do anything except killing background window.

If I want to run in fullscreen, I must use the config file to specify it to launch dosbox in fullscreen mode directly.
Doing this way, I can use ALT+ENTER to use window mode and ALT+ENTER to use fullscreen mode again, without any problem EXCEPT that the window mode size is smaller than "normal" window mode (when launching dosbox without full screen).

I hope this is a useful issue, sorry if a duplicate.

Thanks :)

**Comment by dreamer** [comment]
@Draky50110 thank you for the report!. I had this issue happening once in Fedora 33, but the problem disappeared after a system update, and @kcgen experienced it as well in Ubuntu 20.10 just before we released 0.76.0.

We were not sure if other people will experience this problem as well.

Out of curiosity: does this problem happen with 0.75.2 build as well? Perhaps one of the fixes we implemented broke some assumptions compositors make.

**Comment by kcgen** [comment]
>Out of curiosity: does this problem happen with 0.75.2 build as well

Yup.. it goes way back :-) Definitely an issue w/ Gnome  3.38.1

@Draky50110 , I actually had to give up on Gnome entirely and switch to XFCE.. which I've actually be quite happy with (although I would prefer it to be backed by Wayland only to get better test coverage).

**Comment by Draky50110** [comment]
@kcgen I tested once Wayland as one of the Ubuntu release were using it by default instead of X11 and they switched back to X11 the next release, as it was full of bug (bugs I experienced too) so I prefer to stay with X11 until it's fully bug-free with Ubuntu.
And I like Gnome ;) but in fact this "fullscreen" bug is not very important for me, will see if next release of Gnome will repair it.

**Comment by dreamer** [comment]
For the record: we have other, I think related bugs:

- MangoHud shows up on the splash screen and then disappears when the DOS shell or game appears
- I saw one-two reports of people booting to black screen (happened to me once and then "fixed itself")

So I don't rule out a bug on our side, but I would say it's 90% chance a Gnome (shell? mutter?) problem.

Anyway, this bug resembles https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3227 a little bit.

Can you check if problem is reproducible when starting in pure Wayland (`SDL_VIDEODRIVER=wayland dosbox`)? 
Additional notes for Wayland:
- Wayland does not work with Snap package at all (either compile yourself or use PPA - instructions in: https://dosbox-staging.github.io/downloads/linux/)
- Window will have no decoration (Gnome refuses to implement served-side decorations) - but Alt+Enter should work regardless.

**Comment by kcgen** [comment]
> Can you check if problem is reproducible when starting in pure Wayland (SDL_VIDEODRIVER=wayland dosbox)?

This was a good exercise; thanks for suggesting it @dreamer!

On first launch (with vanilla conf), I get the usual 640x480 console, however it's borderless and appears to be "snapped" to the bottom edge of my browser.  Not a big deal, but the lack of border was interesting.

![2020-12-06_15-03](https://user-images.githubusercontent.com/1557255/101295727-4549f000-37d4-11eb-838a-9fb0b987df39.png)

Pressing **Alt + Enter** pops it up fullscreen! Hey.. that's better than what I had.

![Screenshot from 2020-12-06 15-03-22](https://user-images.githubusercontent.com/1557255/101295730-48dd7700-37d4-11eb-980f-04945512b602.png)

Pressing **Alt + Enter** again to depart fullscreen only gets part way..  but I can easily **Ctrl + F9** to gracefully quit.

![Screenshot from 2020-12-06 15-03-28](https://user-images.githubusercontent.com/1557255/101295733-4bd86780-37d4-11eb-8543-da090258749d.png)

**Comment by dreamer** [comment]
Decoration part is #650

**Comment by grapeli** [comment]
I don't use wayland very often, but in my opinion it may be a regression. I didn't weigh it down before.

A short [video](https://drive.google.com/file/d/1WdN3S3F9ctHdermOWKYZVzkHGp1DU_tZ/view?usp=sharing) of what it looks like under the Sway. Exiting fullscreen does not restore the window to the correct size.

Dosbox-x is behaving fine in this regard. Although it has another drawback. Only the Sway keyboard shortcut can go to full screen properly.

**Comment by dreamer** [comment]
**VulpesFoxnik** on Discord did some testing and confirmed this is Gnome 3.38.1 issue in Ubuntu 20.10. The next update (scheduled early January, I believe should fix this problem for Ubuntu users). On other distros, the problem might be already solved (e.g. I can't reproduce it on Fedora with Gnome 3.38.2).

Setting this to external and let's wait for Ubuntu update before closing it.

**Comment by Draky50110** [comment]
Ok. Fixed :D 
This afternoon, just launched DOSBox and ALT+ENTER gave me fullscreen without any problem (smooth transition), and get back to window mode also without any problem.

Didn't notice but when but I ran an update and now I'm on Gnome-Shell 3.38.**2** :) 
So, in conclusion : fullscreen mode is buggy with Gnome-Shell 3.38.1 and it get corrected with .2 update.
[Full changelog](https://fr2.rpmfind.net/linux/gnome.org/core/3.38/3.38.2/NEWS) if you want to trace the change :P
